packages/back-end/src/models/FeatureRevisionModel.ts

```diff
@@ -302,12 +334,98 @@ export async function markRevisionAsPublished(
   );
 }
 
+export async function markRevisionAsReviewRequested(
+  revision: FeatureRevisionInterface,
+  user: EventAuditUser,
+  comment?: string
+) {
+  const action = "Review Requested";
+
+  const log: RevisionLog = {
+    action,
+    subject: "",
+    timestamp: new Date(),
+    user,
+    value: JSON.stringify(comment ? { comment } : {}),
+  };
+  migrateToPendingReview(revision.organization, revision.featureId);
+  await FeatureRevisionModel.updateOne(
+    {
+      organization: revision.organization,
+      featureId: revision.featureId,
+      version: revision.version,
+    },
+    {
+      $set: {
+        status: "pending-review",
+        publishedBy: user,
+        datePublished: new Date(),
+        dateUpdated: new Date(),
+        comment: comment ?? revision.comment,
+      },
+      $push: {
+        log,
+      },
+    }
+  );
+}
+
+export async function submitReviewAndComments(
+  revision: FeatureRevisionInterface,
+  user: EventAuditUser,
+  reviewSubmittedType: ReviewSubmittedType,
+  comment?: string
+) {
+  const action = reviewSubmittedType;
+  let status = "pending-review";
+  switch (reviewSubmittedType) {
+    case "Approved":
+      status = "approved";
+      break;
+    case "Requested Changes":
+      status = "changes-requested";
+      break;
+    default:
+      status = "pending-review";
+  }
+  const log: RevisionLog = {
+    action,
+    subject: "",
+    timestamp: new Date(),
+    user,
+    value: JSON.stringify(comment ? { comment } : {}),
+  };
+  if (status === "approved") {
+    await migrateToApproved(revision.organization, revision.featureId);
+  }
+
+  await FeatureRevisionModel.updateOne(
+    {
+      organization: revision.organization,
+      featureId: revision.featureId,
+      version: revision.version,
+    },
+    {
+      $set: {
+        status,
+        publishedBy: user,
+        datePublished: new Date(),
```

Comment:
Same here - we should probably not set these fields since nothing is being published here, it's just being reviewed.